spec/unit/network/authconfig_spec.rb

```diff
@@ -107,3 +107,29 @@
     end
   end
 end
+
+describe Puppet::Network::AuthConfig do
+  class TestAuthProvider
+    def initialize(rights=nil); end
+    def check_authorization(method, path, params); end
+  end
+
+  it "instantiates authprovider_class with rights" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    rights = Puppet::Network::Rights.new
+    TestAuthProvider.expects(:new).with(rights)
+    described_class.new(rights)
+  end
+
+  it "delegates authorization check to authprovider_class" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    TestAuthProvider.any_instance.expects(:check_authorization).with(:save, '/path/to/resource', {})
+    described_class.new.check_authorization(:save, '/path/to/resource', {})
+  end
+
+  it "uses DefaultAuthProvider by default" do
+    Puppet::Network::AuthConfig.authprovider_class = nil
```

Comment:
Ugh...I don't like that I have to explicitly "reset" the `authprovider_class` class variable between tests, but I'm not sure how to do make rspec handle this (if it can).

This is why I wrote `AuthConfig` the way I did with the `@@authprovider_class = nil` and then the getter that returns `@@authprovider_class || DefaultAuthProvider`.

Ideally we'd have `@@authprovider_class = DefaultAuthProvider` but then I don't know how to write this test that ensures this default case.